### PR TITLE
Disable http response egress - testing

### PR DESCRIPTION
**Pull request by kkeirstead**
Added in two test cases to ensure the proper error ensues when using the --no-http-egress flag for dumps and logs.

### CODE REVIEW

**Comment by jander-msft** [comment]
@kkeirstead, I think you included your commits from the prior change. I would recommend rebasing from `main` and skipping those commits so that only the test additions are in the PR. You can then force push the change to the remote branch.

File: src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs [hunk 1]
```diff
@@ -211,6 +211,39 @@ public async Task SharedConcurrencyLimitTest()
...
+
+                    // Dump Error Check
+                    ValidationProblemDetailsException validationProblemDetailsExceptionDumps = await Assert.ThrowsAsync<ValidationProblemDetailsException>(
+                        () => appClient.CaptureDumpAsync(appRunner.ProcessId, DumpType.Mini));
+                    Assert.Equal(HttpStatusCode.BadRequest, validationProblemDetailsExceptionDumps.StatusCode);
+                    Assert.Equal(StatusCodes.Status400BadRequest, validationProblemDetailsExceptionDumps.Details.Status);
+
+                    // Logs Error Check
+                    ValidationProblemDetailsException validationProblemDetailsExceptionLogs = await Assert.ThrowsAsync<ValidationProblemDetailsException>(
+                            () => appClient.CaptureLogsAsync(appRunner.ProcessId, TestTimeouts.LogsDuration, Extensions.Logging.LogLevel.None, WebApi.LogFormat.NDJson));
```
Comment (jander-msft) [inline src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs:238]: nit: Add `using`s at the top of this file for `Microsoft.Extensions.Logging` and `Microsoft.Diagnostics.Monitoring.WebApi` and then remove the namespace disambiguation here. It makes it a little harder to read given the length of the code line.

File: src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs [hunk 2]
```diff
@@ -211,6 +211,39 @@ public async Task SharedConcurrencyLimitTest()
...
+                appValidate: async (appRunner, appClient) =>
+                {
+                    ProcessInfo processInfo = await appClient.GetProcessAsync(appRunner.ProcessId);
+                    Assert.NotNull(processInfo);
+
+                    // Dump Error Check
+                    ValidationProblemDetailsException validationProblemDetailsExceptionDumps = await Assert.ThrowsAsync<ValidationProblemDetailsException>(
+                        () => appClient.CaptureDumpAsync(appRunner.ProcessId, DumpType.Mini));
+                    Assert.Equal(HttpStatusCode.BadRequest, validationProblemDetailsExceptionDumps.StatusCode);
+                    Assert.Equal(StatusCodes.Status400BadRequest, validationProblemDetailsExceptionDumps.Details.Status);
```
Comment (jander-msft) [inline src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs:234]: Given that the error message doesn't change between the endpoints or what is calling it, do you think you could add an assert for the expected error message? Maybe you can reference the resource string instead of embedding the literal message.

**Comment by kkeirstead** [comment]
@jander-msft Yep, I noticed that and was able to merge to avoid all the files from appearing as changed but it kept that commit history. I'll see if I can make it look like less of a mess.

**Comment by kkeirstead** [comment]
I had some difficulty with figuring out how to do the rebase, so I ended up taking a more manual approach to addressing the out-of-place commits. This should have the same intended effect of cleaning things up - just a different way of getting there.

File: src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs [hunk 3]
```diff
@@ -211,6 +214,43 @@ public async Task SharedConcurrencyLimitTest()
...
+                    // This should be identical to the error message found in Strings.resx
+                    Assert.Equal("HTTP egress is not enabled.", validationProblemDetailsExceptionDumps.Message);
```
Comment (jander-msft) [inline src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs:239]: Please make this a constant and move the comment to the constant.

File: src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs [hunk 4]
```diff
@@ -24,6 +27,7 @@
 using System.Threading.Tasks;
 using Xunit;
 using Xunit.Abstractions;
+using DiagnosticPortConnectionMode = Microsoft.Diagnostics.Monitoring.UnitTests.Options.DiagnosticPortConnectionMode;
```
Comment (jander-msft) [inline src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs:30]: Why is this necessary? The other existing tests in this file are already using this type.

**Review by jander-msft — CHANGES_REQUESTED** [review]

File: src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs (hunk 4 above)
Comment (kkeirstead) [reply src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs:30]: Since you recommended I add the extension for WebApi, I started getting an error for DiagnosticPortConnectionMode that it was an ambiguous reference between Options and WebApi. We had added this when we had the same issue in LogsTests, so I reimplemented it here. I just pushed changes (without changing that), so if you have something else in mind let me know and I'll push out a new version.

**Comment by kkeirstead** [comment]
Looks like the old commits came back with the way I made my changes; will push again shortly to resolve this.

File: src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs (hunk 4 above)
Comment (jander-msft) [reply src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs:30]: > Since you recommended I add the extension for WebApi, I started getting an error for DiagnosticPortConnectionMode that it was an ambiguous reference between Options and WebApi. We had added this when we had the same issue in LogsTests, so I reimplemented it here. I just pushed changes (without changing that), so if you have something else in mind let me know and I'll push out a new version.

I reread this a few times and didn't understand what you were referring to until I realized that "extension" is referring to the `using Microsoft.Diagnostics.Monitoring.WebApi;` statement. This makes more sense now.

**Review by jander-msft — APPROVED** [review]
